backend/typescript/services/implementations/checkInService.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable class-methods-use-this */
+import { snakeCase } from "lodash";
+import dayjs from "dayjs";
+import ICheckInService from "../interfaces/checkInService";
+import {
+  CheckInDTO,
+  CreateCheckInDTO,
+  UpdateCheckInDTO,
+  DTOTypes,
+} from "../../types";
+import logger from "../../utilities/logger";
+import CheckIn from "../../models/checkIn.model";
+import getErrorMessage from "../../utilities/errorMessageUtil";
+import IEmailService from "../interfaces/emailService";
+import IVolunteerService from "../interfaces/volunteerService";
+import { toSnakeCase } from "../../utilities/servicesUtils";
+
+const Logger = logger(__filename);
+
+class CheckInService implements ICheckInService {
+  emailService: IEmailService | null;
+
+  volunteerService: IVolunteerService | null;
+
+  constructor(
+    emailService: IEmailService | null = null,
+    volunteerService: IVolunteerService | null = null,
+  ) {
+    this.emailService = emailService;
+    this.volunteerService = volunteerService;
+  }
+
+  async createCheckIn(checkIn: CreateCheckInDTO): Promise<Array<CheckInDTO>> {
+    let snakeCaseCheckIns: Array<CheckIn> = [];
+    try {
+      const originalStartDate: Date = new Date(checkIn.startDate);
+      const originalEndDate: Date = new Date(checkIn.endDate);
+
+      let nextStartDate = dayjs(originalStartDate);
+      let nextEndDate = dayjs(originalStartDate);
+
+      nextEndDate = nextEndDate
+        .hour(originalEndDate.getHours())
+        .minute(originalEndDate.getMinutes())
+        .second(originalEndDate.getSeconds())
+        .millisecond(originalEndDate.getMilliseconds());
+
+      const checkInsToBeCreated: DTOTypes[] = [];
```

Comment:
small nit: but can we move this constant to the start of the function for readability